x-pack/legacy/plugins/siem/public/lib/connectors/servicenow.tsx

```diff
@@ -0,0 +1,222 @@
+    []
+  );
+
+  const handleOnBlurSecretConfig = useCallback(
+    (key: string) => {
+      if (['username', 'password'].includes(key) && get(key, action.secrets) == null) {
+        editActionSecrets(key, '');
+      }
+    },
+    [action.secrets]
+  );
+
+  const handleOnChangeMappingConfig = useCallback(
+    (newMapping: CasesConfigurationMapping[]) =>
+      editActionConfig('casesConfiguration', {
+        ...action.config.casesConfiguration,
+        mapping: newMapping,
+      }),
+    [action.config]
+  );
+
+  return (
+    <>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            id="apiUrl"
+            fullWidth
+            error={errors.apiUrl}
+            isInvalid={isApiUrlInvalid}
+            label={i18n.SERVICENOW_API_URL_LABEL}
+          >
+            <EuiFieldText
+              fullWidth
+              isInvalid={isApiUrlInvalid}
+              name="apiUrl"
+              value={apiUrl}
+              data-test-subj="apiUrlFromInput"
+              placeholder="https://<instance>.service-now.com"
+              onChange={handleOnChangeActionConfig.bind(null, 'apiUrl')}
+              onBlur={handleOnBlurActionConfig.bind(null, 'apiUrl')}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            id="username"
+            fullWidth
+            error={errors.username}
+            isInvalid={isUsernameInvalid}
+            label={i18n.SERVICENOW_USERNAME_LABEL}
+          >
+            <EuiFieldText
+              fullWidth
+              isInvalid={isUsernameInvalid}
+              name="username"
+              value={username}
+              data-test-subj="usernameFromInput"
+              onChange={handleOnChangeSecretConfig.bind(null, 'username')}
+              onBlur={handleOnBlurSecretConfig.bind(null, 'username')}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            id="password"
+            fullWidth
+            error={errors.password}
+            isInvalid={isPasswordInvalid}
+            label={i18n.SERVICENOW_PASSWORD_LABEL}
+          >
+            <EuiFieldPassword
+              fullWidth
+              isInvalid={isPasswordInvalid}
+              name="password"
+              value={password}
+              data-test-subj="passwordFromInput"
+              onChange={handleOnChangeSecretConfig.bind(null, 'password')}
+              onBlur={handleOnBlurSecretConfig.bind(null, 'password')}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiSpacer size="xs" />
+          <FieldMapping
+            disabled={false}
+            mapping={mapping as CasesConfigurationMapping[]}
```

Comment:
this is a bit confusing, any way we can do this without typecasting with `as`?

Reply:
I think it can be done when we unify the schema between servicenow action and cases configure API. @XavierM ?

Reply:
Yes that's correct, we said that we will move some in our case common at on point